Library/Homebrew/formula_installer.rb

```diff
@@ -589,9 +589,16 @@ def link(keg)
       keg.remove_linked_keg_record
     end
 
+    link_overwrite_backup = []
+
     begin
       keg.link
     rescue Keg::ConflictError => e
+      if formula.link_overwrite?(e.dst) && !link_overwrite_backup.include?(e.dst)
+        e.dst.rename "#{e.dst}.bak"
```

Comment:
> it'll happen every time you brew upgrade glib-networking

No it will only happen once.

Another reason that we want to backup is we need restore them during unexpected error https://github.com/xu-cheng/homebrew/blob/link/Library/Homebrew/formula_installer.rb#L626-L629